zorg/buildbot/builders/BOLTBuilder.py

```diff
@@ -0,0 +1,69 @@
+from buildbot.plugins import steps
+from zorg.buildbot.commands.CmakeCommand import CmakeCommand
+from zorg.buildbot.builders.UnifiedTreeBuilder import getLLVMBuildFactoryAndSourcecodeSteps, addCmakeSteps, addNinjaSteps
+from zorg.buildbot.process.factory import LLVMBuildFactory
+
+def getBOLTCmakeBuildFactory(
+           clean = False,
+           bolttests = False,
+           extra_configure_args = None,
+           env = None,
+           **kwargs):
+
+    if env is None:
+        env = dict()
+
+    bolttests_dir = "bolt-tests"
+
+    cleanBuildRequested = lambda step: clean or step.build.getProperty("clean", default=step.build.getProperty("clean_obj"))
+
+    checks = ['check-bolt']
+    extra_steps = []
+
+    f = getLLVMBuildFactoryAndSourcecodeSteps(
+            depends_on_projects=['clang', 'lld', 'bolt'],
```

Comment:
Curious: why does bold depends on clang?

Reply:
BOLT itself doesn't, it's the C/C++ tests that come with it. My understanding is that `depends_on_projects` translate into `-DLLVM_ENABLE_PROJECTS`.

Reply:
> My understanding is that depends_on_projects translate into -DLLVM_ENABLE_PROJECTS.

No, this is specified separately. This is about what the bot will be triggered on (i.e. changes to clang itself will trigger new builds), to avoid triggering spurious build.

Reply:
Thanks for clarification! Although our C/C++/assembly tests depend on clang and lld, I think it doesn't make sense to trigger BOLT build/test on commits to these projects. I'll drop this `depends_on_projects` line, to reflect that we only depend on llvm which is added by default.